plugins/modules/firehose/module.go

```diff
@@ -383,6 +390,53 @@ func (m *Module) Act(r *resource.Resource, action string, params map[string]inte
 	return r.Configs, nil
 }
 
+func Log(ctx context.Context, r *resource.Resource, filter map[string]string) (<-chan module.LogChunk, error) {
+	var cfg config
+	if err := mapstructure.Decode(r.Configs, &cfg); err != nil {
+		return nil, errors.New("unable to parse configs")
+	}
+
+	var releaseConfig *helm.ReleaseConfig
+	if err := mapstructure.Decode(r.Configs[releaseConfigString], &releaseConfig); err != nil {
+		return nil, errors.New("unable to parse configs")
+	}
+
+	podLogs, err := kubelogger.GetStreamingLogs(ctx, releaseConfig.Namespace, releaseConfig.Name)
+	if err != nil {
+		return nil, err
+	}
+	logs := make(chan module.LogChunk)
+	defer close(logs)
+	go func() {
+		for {
+			buf := make([]byte, 10000)
+			numBytes, err := podLogs.Read(buf)
+			if numBytes == 0 {
+				break
+			}
+			if err == io.EOF {
+				break
+			}
+
+			if err != nil {
```

Comment:
merge all these if blocks ..

```golang
if err != nil {
   break 
}
```

* why are you exiting on `numBytes == 0`. There might not be new logs but that does not necessarily indicate EOF.
* since `err==io.EOF` & `err != nil` cases are handled exactly the same, why not just have `err != nil` check and break.